src/states/node.rs

```diff
@@ -103,6 +104,8 @@ pub struct Node {
     tick_timer_token: u64,
     timer: Timer,
     tunnels: Tunnels,
+    /// Key is (dst, sections), value is BTreeSet<src>
+    out_of_seq_own_merge: LruCache<(Prefix<XorName>, SectionMap), BTreeSet<Prefix<XorName>>>,
```

Comment:
This is to ensure we don't handle outdated record.
the failing test has the same sections for a matching pair of OwnSectionMerge messages.

We can use `dst` only as key, but then probably needs to narrow the time_out